structuralcodes/codes/mc2010/_concrete_punching.py

```diff
@@ -0,0 +1,579 @@
+    edge_par: bool,
+    edge_per: bool,
+    corner: bool,
+    m_rd: float,
+    m_pd: float,
+    alpha: float,
+    f_bd: float,
+    f_ywk: float,
+    kam_w: float,
+    a_sw: float,
+    gamma_s: float,
+):
+    """The punching resistance from shear reinforcement
+     fib Model Code 2010, eq. (7.3-64) and (7.3-65)
+
+    Args:
+        e_u (float): The ecentrisity of the result of shear forces
+        with respect to the centroid (Figure 7.3-27b)
+        b_u (float): The diamter of a circle with same surface as the
+        region inside the basic control perimeter (Figure 7.3-27b)
+        l_x (float): The distance between two columns in x direction
+        l_y (float): The distance between two columns in y direction
+        f_yd (float): Design strength of reinforment steel in MPa
+        d (float): The mean value of the effective depth in mm
+        e_s (float): The E_s-modulus for steel in Mpa
+        approx_lvl_p (float): The approx level for punching
+        v_ed (float): The acting shear force from the columns
+        l_min (float): The shorter side of the the L_x and L_y
+        inner (bool): Is true only if the column is a inner column
+        edge_par (bool): Is true only if the column is a edge column with
+        tention reinforcement paralell to the edge
+        edge_per (bool): Is true only if the column is a edge column with
+        tention reinforcement perpendicular to the edge
+        corner (bool): Is true only if the column is a corner column
+        m_rd (float): The design average strength per unit length in MPa
+        m_pd: (float): The average decompresstion moment due to prestressing
+        in MPa
+        alpha (float): Inclination of the stirrups in degrees
+        f_bd (float): The design bond strength in MPa
+        f_ywk (float): Characteristic yield strength of the shear
+        reinforcement in Mpa
+        kam_w (float): The diameter of the shear reinforcement
+        a_sw (float): The area of the shear reinforcement in mm^2
+        gamma_s (float): Steels reduction factor
+
+    Return: Punching resistance that comes from reinforcement
+    """
+    f_ywd = f_ywk / gamma_s
+    k_e = 1 / (1 + e_u / b_u)
+    sigma_swd = min(
+        (
+            e_s
+            * psi_punching(
+                l_x,
+                l_y,
+                f_yd,
+                d,
+                e_s,
+                approx_lvl_p,
+                v_ed,
+                e_u,
+                l_min,
+                inner,
+                edge_par,
+                edge_per,
+                corner,
+                m_rd,
+                m_pd,
+            )
+            / 6
+        )
+        * (sin(alpha * pi / 180) + cos(alpha * pi / 180))
+        * (sin(alpha * pi / 180) + f_bd * d / (f_ywd * kam_w)),
+        f_ywd,
+    )
+
+    if (a_sw * k_e * sigma_swd * sin(alpha * pi / 180)) < 0.5 * v_ed:
+        warnings.warn(
+            """In order to ensure sufficent deformation capacity,
```

Comment:
Consider rewriting:
"...the shear resistance in punching most increase." -> "...consider increasing the amount of punching shear reinforcement."

Reply:
Done